apps/src/code-studio/progressRedux.js

```diff
@@ -84,18 +85,51 @@ const initialState = {
 /**
  * Thunks
  */
+
+// The user has navigated to a new level in the current lesson,
+// so we should update the browser and also set this as the new
+// current level.
 export function navigateToLevelId(levelId) {
   return (dispatch, getState) => {
     const state = getState().progress;
-    const newLevel = state.lessons[0].levels.find(level =>
-      level.ids.find(id => id === levelId)
-    );
+    const newLevel = state.lessons
+      .find(lesson => lesson.id === state.currentLessonId)
+      .levels.find(level => level.ids.find(id => id === levelId));
 
     updateBrowserForLevelNavigation(state, newLevel.url, levelId);
     dispatch(setCurrentLevelId(levelId));
   };
 }
 
+// The user has successfully completed the level and the page
+// will not be reloading.
+export function sendSuccessReport(appType) {
+  return (dispatch, getState) => {
+    const state = getState().progress;
+    const levelId = state.currentLevelId;
+    const currentLevel = state.lessons
+      .find(lesson => lesson.id === state.currentLessonId)
+      .levels.find(level => level.ids.find(id => id === levelId));
+    const scriptLevelId = currentLevel.id;
+
+    const data = {
+      app: appType,
+      result: true,
+      testResult: 100,
```

Comment:
For my understanding, anything special about this number? Is it always 100 whenever a milestone is completed?

Reply:
It's an "ideal pass".  I'll update with the official constant from [here](https://github.com/code-dot-org/code-dot-org/blob/49438f119d0551b5aa4c23b0f4860b395696a463/apps/src/constants.js#L100), which corresponds to the server-side constant [here](https://github.com/code-dot-org/code-dot-org/blob/49438f119d0551b5aa4c23b0f4860b395696a463/lib/cdo/activity_constants.rb#L17).